packages/blade/src/components/Input/DropdownInputTriggers/AutoComplete.tsx

```diff
@@ -0,0 +1,198 @@
+  }, [globalFilteredValues.length]);
+
+  const onInputValueChange: BaseDropdownInputTriggerProps['onInputValueChange'] = ({
+    name,
+    value,
+  }) => {
+    setInputValue(value ?? '');
+    props.onInputValueChange?.({ name, value });
+    setActiveTagIndex(-1);
+
+    if (!isOpen) {
+      setIsOpen(true);
+    }
+
+    if (!props.filteredValues) {
+      // eslint-disable-next-line no-lonely-if
+      if (value && options && options.length > 0) {
+        const filteredOptions = getOptionValues().filter((optionValue) =>
+          optionValue.toLowerCase().startsWith(value.toLowerCase()),
+        );
+        setGlobalFilteredValues(filteredOptions);
+      } else {
+        setGlobalFilteredValues(getOptionValues());
+      }
+    }
+  };
+
+  const onTriggerKeydown: BaseDropdownInputTriggerProps['onTriggerKeydown'] = (e) => {
+    if (e.key === 'Enter') {
+      setActiveTagIndex(-1);
+      setGlobalFilteredValues(getOptionValues());
+    } else if (
+      e.key === 'Backspace' &&
+      !inputValue &&
+      activeTagIndex < 0 &&
+      selectedIndices.length > 0
+    ) {
+      if (isControlled) {
+        setControlledValueIndices(selectedIndices.slice(0, -1));
+      } else {
+        setSelectedIndices(selectedIndices.slice(0, -1));
+      }
+    }
+    onBaseDropdownInputKeydown?.(e);
+  };
+
+  const onSelectionChange: BaseDropdownInputTriggerProps['onChange'] = ({ values }) => {
+    if (selectionType === 'multiple') {
+      setInputValue('');
+      props.onInputValueChange?.({ name: props.name, value: '' });
+      setActiveTagIndex(-1);
+      setGlobalFilteredValues(getOptionValues());
+    } else {
+      const displayText = options.find((option) => option.value === values[0])?.title;
+      props.onInputValueChange?.({
+        name: props.name,
+        value: displayText,
+      });
+      setInputValue(displayText ?? '');
+    }
+    props.onChange?.({ name: props.name, values });
+  };
+
+  return {
+    onSelectionChange,
+    onTriggerKeydown,
+    onInputValueChange,
+  };
+};
+
+const _AutoComplete = (
+  props: AutoCompleteProps,
+  ref: React.ForwardedRef<BladeElementRef>,
+): React.ReactElement => {
+  const [uncontrolledInputValue, setInputValue] = React.useState('');
+  const inputValue = props.inputValue ?? uncontrolledInputValue;
+
+  const {
+    isOpen,
+    options,
+    setFilteredValues: setGlobalFilteredValues,
+    hasAutoCompleteInBottomSheetHeader,
+    setHasAutoCompleteInBottomSheetHeader,
+    onTriggerClick,
+    dropdownTriggerer,
+  } = useDropdown();
+
+  const getOptionValues = React.useCallback(() => {
+    return options.map((option) => option.value);
+  }, [options]);
+
+  const { onSelectionChange, onTriggerKeydown, onInputValueChange } = useAutoCompleteHandlers({
+    props,
+    inputValue,
+    setInputValue,
+    getOptionValues,
+  });
+
+  React.useEffect(() => {
+    if (dropdownTriggerer !== 'AutoComplete') {
+      // When AutoComplete is mounted but not as trigger,
```

Comment:
Thats a big assumption, no? Can we not use Bottomsheet context to check this?

Reply:
This is actually simpler logic than having to check React.Children of header. Also not a big assumption since it does what it is expected to do. Set a flag when AutoComplete is not used as a trigger. This flag is used to disable enable some trigger properties on that AutoComplete then. As far as we don't decide to show AutoComplete as trigger and inside header both, I can't think of scenario where this breaks 🤔  


We'll have to go into checking the children with React.Children then. Wanted to avoid that wherever possible.

Also this will kind of allow you to put AutoComplete anywhere inside of the Dropdown so in future if we decide to AutoComplete in DropdownHeader, it can work on this logic.

Reply:
rephrased the comment a bit